geoopt/manifolds/product.py

```diff
@@ -435,3 +437,96 @@ def origin(
             )
         tensor = self.pack_point(*points)
         return geoopt.ManifoldTensor(tensor, manifold=self)
+
+
+class StereographicProductManifold(ProductManifold):
+    """
+    Product Manifold for Stereographic manifolds.
+
+    Examples
+    --------
+    A Torus
+
+    >>> import geoopt
+    >>> sphere = geoopt.SphereProjection()
+    >>> torus = StereographicProductManifold((sphere, 2), (sphere, 2))
+    """
+
+    def __init__(
+        self,
+        *manifolds_with_shape: Tuple[Stereographic, Union[Tuple[int, ...], int]],
+    ):
+        super().__init__(*manifolds_with_shape)
+
+    def expmap0(self, u: torch.Tensor) -> torch.Tensor:
+        target_batch_dim = _calculate_target_batch_dim(u.dim())
+        mapped_tensors = []
+        for i, manifold in enumerate(self.manifolds):
+            tangent = self.take_submanifold_value(u, i)
+            mapped = manifold.expmap0(tangent)
+            mapped = mapped.reshape((*mapped.shape[:target_batch_dim], -1))
+            mapped_tensors.append(mapped)
+        return self.pack_point(*mapped_tensors)
+
+    def dist2plane(
```

Comment:
Need to figure out scaling requirements here. Should be similar to how scaling is declared in Stereographic manifold

Reply:
In short scaling is used to rescale tangent space coordinates